metals/src/main/scala/scala/meta/internal/metals/formatting/MultilineString.scala

```diff
@@ -87,10 +88,44 @@ case class MultilineString(userConfig: () => UserConfiguration)
     if (lastQuote != -1) Some((lastQuote, quoteClosed)) else None
   }
 
+  private def getIndexOfLastOpenTripleQuote(
+      closedFromPreviousLines: Boolean,
+      line: String,
+  ): Option[(Int, Boolean)] = {
+    var lastTripleQuote = -1
+    var escaped = false
+    var tripleQuoteClosed = closedFromPreviousLines
+    var quoteNum = 0
+    for (i <- 0 until line.size) {
+      val char = line(i)
+      if (char == '"') {
+        if (!escaped) {
+          quoteNum = quoteNum + 1
+          if (quoteNum == 3) {
+            lastTripleQuote = i
+            tripleQuoteClosed = !tripleQuoteClosed
+            quoteNum = 0
+          }
+        } else {
+          escaped = !escaped
+        }
+      } else if (char == '\\') {
+        quoteNum = 0
+        escaped = !escaped
```

Comment:
I still don't think escape handling is not needed because of https://github.com/scalameta/metals/pull/4151#discussion_r921969587

Reply:
we are traversing character by character. What if the escape is at the beginning?

Reply:
`\"""`

Reply:
` val a = """ This is a sample \""" quoted string in which escaping matters

Reply:
Ah, right

Reply:
turns out that you were right @tanishiking 

In multiline strings, one cannot escape `"` using `\` character.

Reply:
https://scastie.scala-lang.org/vzmerr/tPvC34B4RM2EIywhJlcLRg/46

Reply:
https://scastie.scala-lang.org/vzmerr/tPvC34B4RM2EIywhJlcLRg/47

Reply:
Thank you for investigating that!